Mage.Sets/src/mage/cards/t/TriumphOfSaintKatherine.java

```diff
@@ -0,0 +1,165 @@
+package mage.cards.t;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.DiesSourceTriggeredAbility;
+import mage.abilities.common.delayed.ReflexiveTriggeredAbility;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.CostImpl;
+import mage.abilities.effects.common.DoWhenCostPaid;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.keyword.LifelinkAbility;
+import mage.abilities.keyword.MiracleAbility;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.game.ExileZone;
+import mage.game.Game;
+import mage.players.Player;
+
+/**
+ * @author sprangg
+ */
+public final class TriumphOfSaintKatherine extends CardImpl {
+
+    public TriumphOfSaintKatherine(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{4}{W}");
+
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.WARRIOR);
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(5);
+        
+        // Lifelink
+        this.addAbility(LifelinkAbility.getInstance());
+
+        // Praesidium Protectiva — When Triumph of Saint Katherine dies, exile it and the top six cards of your library in a face-down pile.
+        // If you do, shuffle that pile and put it back on top of your library.
+        ReflexiveTriggeredAbility ability = new ReflexiveTriggeredAbility(
+                new TriumphOfSaintKatherineEffect().setText("shuffle that pile and put it back on top of your library"), false);
+        this.addAbility(new DiesSourceTriggeredAbility(new DoWhenCostPaid(
+                ability, new TriumphOfSaintKatherineCost(), null, false
+        )).withFlavorWord("Praesidium Protectiva"));
```

Comment:
this isn't a reflexive trigger, it's just an effect that should be done all in one shot